nrt/data/__init__.py

```diff
@@ -86,18 +87,274 @@ def mre_crit_table():
+    around the peak of the simulated variable. Actual vegetation signal is often more
+    asymetrical and irregular.
+
+    Args:
+        dates (array-like): List or array of dates (numpy.datetime64)
+        break_idx (int): Breakpoint index in the date array provided. Defaults to
+            ``-1``, corresponding to a stable time-series
+        intercept (float): Intercept of the time-series
+        amplitude (float): Amplitude of the harmonic model (note that at every point
+            of the time-series, the actual model amplitude is multiplied by the intercept
+        magnitude (float): Break magnitude (always a drop in y value)
+        recovery_time (int): Time (in days) to recover the initial intersect value
+            following a break
+        sigma_noise (float): Sigma value of the normal distribution (mean = 0) from which
+            noise values are drawn
+        n_outlier (int): Number of outliers randomly assigned to observations of the
+            time-series
+        outlier_value (float): Value to assign to outliers
+        n_nan (int): Number of ``np.nan`` (no data) assigned to observations of the
+            time-series
+
+    Example:
+        >>> from nrt import data
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+
+        >>> dates = np.arange('2018-01-01', '2022-06-15', dtype='datetime64[W]')
+        >>> ts = data.make_ts(dates=dates, break_idx=30)
+
+        >>> plt.plot(dates, ts)
+        >>> plt.show()
+
+    Returns:
+        np.ndarray: Array of simulated values of same size as ``dates``
+    """
+    timestamps = (dates - np.datetime64(0, 'D')).astype(int)
+    ydays = (dates - dates.astype('datetime64[Y]')).astype(int) + 1
+    y = np.empty_like(dates, dtype=np.float64)
+    # INtercept array
+    y[:] = intercept
+    # Build trend segment if break
+    if break_idx != -1:
+        # Segment bounds
+        segment_start_y = intercept - magnitude
+        segment_start_timestamp = timestamps[break_idx]
+        segment_end_timestamp = segment_start_timestamp + recovery_time
+        segment_end_idx = np.abs(segment_end_timestamp - timestamps).argmin()
+        # Compute y values
+        recovery_rate = magnitude / recovery_time
+        days_since_break = timestamps - segment_start_timestamp
+        trend_segment = (recovery_rate * days_since_break + segment_start_y)[break_idx + 1:segment_end_idx + 1]
+        # include into y
+        y[break_idx + 1:segment_end_idx + 1] = trend_segment
+    # Seasonality
+    amplitude_values = amplitude * y
+    season = amplitude * np.sin(2 * np.pi * timestamps / 365.25 - 2)
+    # noise and outliers
+    noise = np.random.normal(0, sigma_noise, dates.size)
+    # Combine the 3 (trend, season, noise) components
+    ts = y + season + noise
+    # Add optional outliers and Nans
+    outliers_idx = np.random.choice(np.arange(0, dates.size), size=n_outlier)
```

Comment:
Might make sense to set `replace=False` here

Reply:
Good point :+1: